js/src/jupyter-gmaps.js

```diff
@@ -183,17 +184,26 @@ export const BaseMarkerView = widgets.WidgetView.extend({
             ...styleOptions
         }
         this.marker = new google.maps.Marker(markerOptions)
+
+        this.infoWindow=new google.maps.InfoWindow({
+            content: infoBoxHtml
+        });
         this.modelEvents()
     },
 
     addToMapView(mapView) {
-        this.marker.setMap(mapView.map)
+        let marker= this.marker;
+        let infoWindow=this.infoWindow;
+        marker.setMap(mapView.map);
+        marker.addListener('click', function() {
+            infoWindow.open(mapView.map, marker);
+        });
     },
 
     modelEvents() {
         // Simple properties:
         const properties = [
-            ['title', 'hover_text']
+            ['title', 'hover_text','info_html']
```

Comment:
This isn't quite right. It should be:

```javascript
    modelEvents() {
        // Simple properties:
        const properties = [
            ['title', 'hover_text']
        ]
        properties.forEach(([nameInView, nameInModel]) => {
            const callback = (
                () => {
                  this.marker.set(
                  nameInView, this.model.get(nameInModel))
                }
            )
            this.model.on(`change:${nameInModel}`, callback, this)
        })

        const infoBoxProperties = [
            ['content', 'info_html']
            // 'content' is the name of the property for the InfoWindow object,
            // 'info_html' is the name of the property in the Python widget
        ]
        infoBoxProperties.forEach(([nameInView, nameInModel]) => {
            const callback = (
                () => {
                    this.infoWindow.set(
                    nameInView, this.model.get(nameInModel))
                }
            )
            this.model.on(`change:${nameInModel}`, callback, this)
        })

        this.setStyleEvents()
    }
```

The idea is that, every time the property `info_html` is changed at the Python level, this triggers an event in JS called `change:info_html` (by the magic of traitlets and ipywidgets). We need to react to that event by changing the property in the google maps `InfoWindow` object.

This lets us change the info text without having to redraw the entire map:

<img width="1112" alt="screen shot 2016-12-11 at 14 19 02" src="https://cloud.githubusercontent.com/assets/1392879/21080669/1b310b44-bfad-11e6-80ef-b6b8ba9e955b.png">

Ultimately, listening to events and handling them correctly should let people build mini-applications running in Jupyter notebooks.